modules/reindex/src/test/java/org/opensearch/index/reindex/ReindexRenamedSettingTests.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.reindex;
+
+import org.junit.Before;
+import org.opensearch.common.settings.Setting;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.test.OpenSearchTestCase;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * A unit test to validate the former name of the setting 'reindex.remote.allowlist' still take effect,
+ * after it is deprecated, so that the backwards compatibility is maintained.
+ * The test can be removed along with removing support of the deprecated setting.
+ */
+public class ReindexRenamedSettingTests extends OpenSearchTestCase {
+    ReindexPlugin plugin;
+
+    @Before
+    public void setup() {
+        this.plugin = new ReindexPlugin();
+    }
```

Comment:
Nitpick, but I think you can replace all this with just:

```
private final ReindexPlugin plugin = new ReindexPlugin();
```

There's generally no need for a `@Before` method unless you need more complex logic than field assignment. (One note about JUnit that often trips people up...the test framework creates a brand new instance of the test class before each individual test run. That mean no instance state is kept between test runs, which is why the simple inline assignment functions the same as assigning fields in a `@Before` method.)

Reply:
Thank you so much for sharing this knowledge to me! 👍👍 I totally had no idea with it. 😲 I will make the change.

Reply:
JUnit5 allows you to [specify the test instance lifecycle],(https://junit.org/junit5/docs/current/user-guide/#writing-tests-test-instance-lifecycle), though the create-new-instance-for-each-test-method behavior is the default and matches the behavior of all previous versions of JUnit.

Reply:
Thanks a lot for telling me the background theory! Hope more people notice it. 😄 I see, since the default lifecycle is "per method", so any class variable will be initialized before calling any method. 
(Though in this file, the variable only used in one method 😅)